tests/functional/test_new_resolver.py

```diff
@@ -27,6 +30,15 @@ def assert_not_installed(script, *args):
         "{!r} contained in {!r}".format(args, installed)
 
 
+def assert_editable(script, *args):
+    # This simply checks whether all of the listed packages have a
+    # corresponding .egg-link file installed.
+    # TODO: Implement a more rigorous way to test for editable installations.
+    egg_links = set("{}.egg-link".format(arg) for arg in args)
+    assert egg_links <= set(os.listdir(script.site_packages_path)), \
+        "{!r} not all found in {!r}".format(args, script.site_packages_path)
```

Comment:
IIRC, other tests use the list of created files returned in the `script.pip()` result object, rather than doing a direct `os.listdir`. But yes, this is something to deal with later, writing a proper "was this installed as editable" test helper.